src/oracle_implementations/discount_rate/NotionalFinance/NotionalFinanceValueProvider.sol

```diff
@@ -16,31 +16,31 @@ contract NotionalFinanceValueProvider is Oracle, Convert {
     int256 internal constant SECONDS_PER_YEAR = 31104000 * 1e18;
 
     address public immutable notionalView;
-    uint16 public immutable currencyId;
+    uint256 public immutable currencyId;
     uint256 public immutable maturityDate;
     uint256 public immutable settlementDate;
 
-    uint256 private immutable oracleRateDecimals;
+    uint256 private immutable lastImpliedRateDecimals;
 
     /// @notice Constructs the Value provider contracts with the needed Notional contract data in order to
     /// calculate the annual rate.
     /// @param timeUpdateWindow_ Minimum time between updates of the value
     /// @param notionalViewContract_ The address of the deployed notional view contract.
     /// @param currencyId_ Currency ID(eth = 1, dai = 2, usdc = 3, wbtc = 4)
-    /// @param oracleRateDecimals_ Precision of the Notional Market rate.
+    /// @param lastImpliedRateDecimals_ Precision of the Notional Market rate.
     /// @param maturity_ Maturity date.
     /// @param settlementDate_ Settlement date.
     constructor(
         // Oracle parameters
         uint256 timeUpdateWindow_,
         // Notional specific parameters
         address notionalViewContract_,
-        uint16 currencyId_,
-        uint256 oracleRateDecimals_,
+        uint256 currencyId_,
```

Comment:
@amarcu It's probably a good idea to add a check that validates if the passed variable is `<= type(uint16).max`.